### PR TITLE
Tw 428 address book

### PR DESCRIPTION
**Pull request by herkoss**

### CODE REVIEW

File: src/interfaces/account.interface.ts [hunk 1]
```diff
@@ -4,11 +4,14 @@ import { LoadableEntityState } from '../store/types';
 import { AccountTokenInterface } from '../token/interfaces/account-token.interface';
 import { ActivityGroup } from './activity.interface';
 
-export interface AccountInterface {
+export interface IAccountBase {
```
Comment (unixvb) [inline src/interfaces/account.interface.ts:7]: follow the same  name structure as it is in other places (for ex `AccountInterface`)

File: src/components/account-dropdown/account-dropdown-item/account-dropdown-item.tsx [hunk 2]
```diff
@@ -18,7 +18,7 @@ import { AccountDropdownItemProps } from './account-dropdown-item.interface';
 import { useAccountDropdownItemStyles } from './account-dropdown-item.styles';
 
 export const AccountDropdownItem: FC<AccountDropdownItemProps> = ({
-  account = emptyAccount,
+  account = { name: '', publicKeyHash: '' },
```
Comment (unixvb) [inline src/components/account-dropdown/account-dropdown-item/account-dropdown-item.tsx:21]: create `emptyAccountBase` and reuse it

File: src/components/dropdown/section-dropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+type DropdownListItemComponent<T> = FC<{
+  item: T;
+  isSelected: boolean;
+}>;
+
+type DropdownActionButtonsComponent = FC<{
+  onPress: EmptyFn;
+}>;
+
+const DropdownComponent = <T extends unknown>({
```
Comment (unixvb) [inline src/components/dropdown/section-dropdown.tsx:52]: `SectionDropdownComponent`

File: src/components/dropdown/section-dropdown.tsx [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  setSearchValue = emptyFn,
+  equalityFn,
+  renderValue,
+  renderListItem,
+  renderActionButtons = emptyComponent,
+  keyExtractor,
+  onValueChange,
+  onLongPress
+}: SectionDropdownProps<T> & SectionDropdownValueProps<T>) => {
+  const [ref, setRef] = useState<SectionList<T, { title: string; data: Array<T> }> | null>(null);
```
Comment (unixvb) [inline src/components/dropdown/section-dropdown.tsx:68]: create type for `{ title: string; data: Array<T> }` and reuse it

File: src/components/dropdown/section-dropdown.tsx [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+  const styles = useDropdownStyles();
+  const dropdownBottomSheetController = useBottomSheetController();
+  const contentHeight = 0.7 * useWindowDimensions().height;
```
Comment (unixvb) [inline src/components/dropdown/section-dropdown.tsx:71]: create hook `useDropdownHeight` and reuse this code

File: src/components/dropdown/section-dropdown.tsx [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+        {renderValue({ value, disabled })}
+      </TouchableOpacity>
+
+      <BottomSheet description={description} contentHeight={contentHeight} controller={dropdownBottomSheetController}>
+        <View style={styles.contentContainer}>
+          {isSearchable && <SearchInput placeholder="Search" onChangeText={setSearchValue} />}
+          <SectionList
+            sections={list}
+            ref={ref => setRef(ref)}
+            getItemLayout={(_, index) => ({ length: itemHeight, offset: itemHeight * index, index })}
```
Comment (unixvb) [inline src/components/dropdown/section-dropdown.tsx:141]: use `createGetItemLayout` and add it into `DropdownComponent`

File: src/components/bottom-sheet/bottom-sheet.tsx [hunk 7]
```diff
@@ -21,13 +21,15 @@ interface Props extends BottomSheetControllerProps {
   title?: string;
   description: string;
   cancelButtonText?: string;
+  onClosePress?: EmptyFn;
```
Comment (unixvb) [inline src/components/bottom-sheet/bottom-sheet.tsx:24]: you have different naming: `**cancel**ButtonText` and `on**Close**Press`

File: src/hooks/use-delete-contact.hook.ts [hunk 8]
```diff
@@ -0,0 +1,22 @@
+import { Alert } from 'react-native';
+import { useDispatch } from 'react-redux';
+
+import { IAccountBase } from '../interfaces/account.interface';
+import { deleteContactAction } from '../store/contacts/contacts-actions';
+
+export const useDeleteContactHandler = (contact: IAccountBase) => {
```
Comment (unixvb) [inline src/hooks/use-delete-contact.hook.ts:7]: why did you move this into separate hook?  
looks like it's not reused

File: src/interfaces/account.interface.mock.ts [hunk 9]
```diff
@@ -1,6 +1,11 @@
 import { AccountTypeEnum } from '../enums/account-type.enum';
 import { mockAccountCredentials } from '../mocks/account-credentials.mock';
-import { AccountInterface } from './account.interface';
+import { AccountInterface, IAccountBase } from './account.interface';
+
+export const mockHdAccountBase: IAccountBase = {
+  name: 'Mock HD Account',
```
Comment (unixvb) [inline src/interfaces/account.interface.mock.ts:6]: it's not HD account, it's cotact

File: src/modals/contact-modals/add-contact-modal.tsx [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+  const onSubmit = (contact: IAccountBase) => {
+    dispatch(addContactAction(contact));
+    goBack();
+  };
+
+  return (
+    <Formik
+      validateOnBlur
+      validateOnChange
+      initialValues={{ name: params?.name ?? '', publicKeyHash: params?.publicKeyHash ?? '' }}
```
Comment (unixvb) [inline src/modals/contact-modals/add-contact-modal.tsx:39]: move this into const

File: src/modals/contact-modals/edit-contact-modal.tsx [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+  const onSubmit = (contact: IAccountBase) => {
+    dispatch(editContactAction(contact));
+    goBack();
+  };
+
+  return (
+    <Formik
+      validateOnBlur
+      validateOnChange
+      initialValues={{ name, publicKeyHash }}
```
Comment (unixvb) [inline src/modals/contact-modals/edit-contact-modal.tsx:37]: the same

File: src/modals/send-modal/send-modal.tsx [hunk 12]
```diff
@@ -60,11 +61,7 @@ export const SendModal: FC = () => {
     [tezosToken, filteredAssetsList]
   );
 
-  const ownAccountsReceivers = useMemo(
-    () => visibleAccounts.filter(({ publicKeyHash }) => publicKeyHash !== selectedAccount.publicKeyHash),
-    [visibleAccounts, selectedAccount.publicKeyHash]
-  );
-  const transferBetweenOwnAccountsDisabled = ownAccountsReceivers.length === 0;
+  const transferDisabled = filteredReceiversList.length === 0;
```
Comment (unixvb) [inline src/modals/send-modal/send-modal.tsx:64]: `transferDisabled` definetly has wrong name

File: src/modals/send-modal/send-modal.tsx [hunk 13]
```diff
@@ -75,10 +72,10 @@ export const SendModal: FC = () => {
         amount: undefined
       },
       receiverPublicKeyHash: initialRecieverPublicKeyHash,
-      ownAccount: ownAccountsReceivers[0],
+      ownAccount: filteredReceiversList[0].data[0],
```
Comment (unixvb) [inline src/modals/send-modal/send-modal.tsx:75]: rename `ownAccount` field

File: src/modals/send-modal/send-modal.tsx [hunk 14]
```diff
@@ -75,10 +72,10 @@ export const SendModal: FC = () => {
...
       transferBetweenOwnAccounts: false
     }),
-    [filteredAssetsListWithTez, ownAccountsReceivers]
+    [filteredAssetsListWithTez]
```
Comment (unixvb) [inline src/modals/send-modal/send-modal.tsx:78]: why don't you have `filteredReceiversList` in deps anymore?

File: src/screens/wallet/wallet.tsx [hunk 15]
```diff
@@ -65,6 +82,23 @@ export const Wallet = () => {
...
+        description={addContactRequest}
+        cancelButtonText="Now now"
+        contentHeight={formatSize(180)}
+        controller={bottomSheetController}
+        onClosePress={handleClosePress}
+      >
+        <BottomSheetActionButton
+          title="Add address"
+          onPress={() => {
+            navigate(ModalsEnum.AddContact, { name: '', publicKeyHash: addContactRequest });
```
Comment (unixvb) [inline src/screens/wallet/wallet.tsx:97]: `addContactRequest` has a bad name, it very noticable here `publicKeyHash: addContactRequest`

File: src/store/contacts/contacts-reducers.ts [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+  editContactAction
+} from './contacts-actions';
+import { contactsInitialState, ContactsState } from './contacts-state';
+
+export const contactsReducers = createReducer<ContactsState>(contactsInitialState, builder => {
+  builder.addCase(addContactAction, (state, { payload }) => {
+    state.contacts.push(payload);
+  });
+  builder.addCase(editContactAction, (state, { payload }) => {
+    const indexContactToEdit = state.contacts.findIndex(contact => contact.publicKeyHash === payload.publicKeyHash);
```
Comment (unixvb) [inline src/store/contacts/contacts-reducers.ts:17]: wrong words order `contactToEditIndex` or simply `contactIndex`

File: src/store/contacts/contacts-reducers.ts [hunk 17]
```diff
@@ -0,0 +1,32 @@
...
+import {
+  addBlacklistedContactAction,
+  addContactAction,
+  addContactRequestAction,
+  deleteContactAction,
+  editContactAction
+} from './contacts-actions';
+import { contactsInitialState, ContactsState } from './contacts-state';
+
+export const contactsReducers = createReducer<ContactsState>(contactsInitialState, builder => {
```
Comment (unixvb) [inline src/store/contacts/contacts-reducers.ts:12]: we are using an immutable state change approach instead of it's mutation.  
It's bad to combine those 2 approaches at the same time (doc says: [` Immer expects that you will either mutate the existing state, or construct a new state value yourself and return it, but not both in the same function! `)](https://redux-toolkit.js.org/usage/immer-reducers#mutating-and-returning-state).  
  
So, it's better to use single approach right now

File: src/store/contacts/contacts-state.ts [hunk 18]
```diff
@@ -0,0 +1,16 @@
...
+  addContactRequest: string;
+}
+export const contactsInitialState: ContactsState = {
+  contacts: [],
+  blacklistedAddresses: [],
+  addContactRequest: ''
+};
+
+export interface ContactsRootState {
+  contacts: ContactsState;
```
Comment (unixvb) [inline src/store/contacts/contacts-state.ts:15]: call this `ContactBookRootState` and change key to `contactBook` to prevent conflict in naming with it's inner `contacts: []`

File: src/store/contacts/contacts-state.ts [hunk 19]
```diff
@@ -0,0 +1,16 @@
+import { IAccountBase } from '../../interfaces/account.interface';
+
+export interface ContactsState {
+  contacts: Array<IAccountBase>;
+  blacklistedAddresses: Array<string>;
+  addContactRequest: string;
+}
+export const contactsInitialState: ContactsState = {
+  contacts: [],
+  blacklistedAddresses: [],
```
Comment (unixvb) [inline src/store/contacts/contacts-state.ts:10]: call it just `ignored` adresses

Comment (unixvb) [reply src/store/contacts/contacts-state.ts:10]: cuz `blacklisted` means we are fully blocking them

File: src/store/root-state.mock.ts [hunk 20]
```diff
@@ -22,5 +23,6 @@ export const mockRootState: RootState = {
   exolix: mockExolixState,
   advertising: mockAdvertisingState,
   market: mockMarketState,
-  notifications: mockNotificationsState
+  notifications: mockNotificationsState,
+  contactBook: mockContactsState
```
Comment (unixvb) [inline src/store/root-state.mock.ts:27]: `mockContactBookState`

File: src/store/create-store.ts [hunk 21]
```diff
@@ -72,7 +75,8 @@ const rootReducer = rootStateReducer<RootState>({
   exolix: exolixReducers,
   advertising: advertisingReducers,
   market: marketReducers,
-  notifications: notificationsReducers
+  notifications: notificationsReducers,
+  contactBook: contactsReducers
```
Comment (unixvb) [inline src/store/create-store.ts:79]: `contactBookReducers`

File: src/store/contacts/contacts-state.ts [hunk 22]
```diff
@@ -0,0 +1,16 @@
...
+  ignoredAddresses: Array<string>;
+  contactCandidatePkh: string;
+}
+export const contactsInitialState: ContactsState = {
+  contacts: [],
+  ignoredAddresses: [],
+  contactCandidatePkh: ''
+};
+
+export interface ContactsBookRootState {
```
Comment (unixvb) [inline src/store/contacts/contacts-state.ts:14]: check all file names inside `src/store/contacts/` - should be "contact book` right now

File: src/store/contacts/contacts-selectors.ts [hunk 23]
```diff
@@ -0,0 +1,7 @@
+import { useSelector } from '../selector';
+
+export const useContactsSelector = () => useSelector(state => state.contactBook.contacts);
+export const useContactsAddressesSelector = () =>
+  useSelector(state => state.contactBook.contacts.map(({ publicKeyHash }) => publicKeyHash));
+export const useContactCandidatePkhSelector = () => useSelector(state => state.contactBook.contactCandidatePkh);
```
Comment (unixvb) [inline src/store/contacts/contacts-selectors.ts:6]: we don't use acronyms in our code as it could cause confusion. fix this

**Review by unixvb — APPROVED** [review]

File: src/store/contact-book/contact-book-reducers.ts [hunk 24]
```diff
@@ -14,13 +14,9 @@ export const contactBookReducers = createReducer<ContactBookState>(contactBookIn
...
   builder.addCase(editContactAction, (state, { payload }) => {
-    const contactIndex = state.contacts.findIndex(contact => contact.publicKeyHash === payload.publicKeyHash);
-
-    if (contactIndex !== -1) {
-      const contactsCopy = [...state.contacts];
-      contactsCopy[contactIndex] = payload;
-      state.contacts = contactsCopy;
-    }
+    const contactsCopy = [...state.contacts];
+    contactsCopy[payload.index] = payload.contact;
```
Comment (unixvb) [inline src/store/contact-book/contact-book-reducers.ts:18]: why do you pass `index` everywhere?  `contact.publicKeyHash` is not unique?

File: src/components/account-dropdown/account-dropdown-item/account-dropdown-item.tsx [hunk 25]
```diff
@@ -24,7 +27,14 @@ export const AccountDropdownItem: FC<AccountDropdownItemProps> = ({
...
   const styles = useAccountDropdownItemStyles();
-  const tezosToken = useTezosTokenSelector(account.publicKeyHash);
+  const selectedRpcUrl = useSelectedRpcUrlSelector();
+  const [tezosBalance, setTezosBalance] = useState('0');
+
+  useEffect(() => {
+    void createReadOnlyTezosToolkit(selectedRpcUrl, readOnlySignerAccount)
+      .tz.getBalance(account.publicKeyHash)
+      .then(value => setTezosBalance(value.toFixed()));
+  }, [selectedRpcUrl, account.publicKeyHash]);
```
Comment (unixvb) [inline src/components/account-dropdown/account-dropdown-item/account-dropdown-item.tsx:37]: move this logic into redux state and load data via epic

File: src/store/contact-book/contact-book-reducers.ts (hunk 24 above)
Comment (herkoss) [reply src/store/contact-book/contact-book-reducers.ts:18]: In case of editing it is not unique

**Review by unixvb — APPROVED** [review]

**Review by unixvb — APPROVED** [review]

**Review by unixvb — APPROVED** [review]

File: ios/TempleWallet.xcodeproj/project.pbxproj [hunk 26]
```diff
@@ -539,11 +539,10 @@
 			buildSettings = {
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				CLANG_ENABLE_MODULES = YES;
-				CODE_SIGN_IDENTITY = "iPhone Distribution";
-				CODE_SIGN_STYLE = Manual;
+				CODE_SIGN_IDENTITY = "Apple Development";
```
Comment (unixvb) [inline ios/TempleWallet.xcodeproj/project.pbxproj:542]: revert changes in this file

**Review by unixvb — APPROVED** [review]

**Review by IrynaKhyzhynska — APPROVED** [review]
android done: 1.13_554

**Review by NovakVlad2424 — APPROVED** [review]
ios done: 1.13_554
